core/shared/src/main/scala/org/virtuslab/yaml/YamlDecoder.scala

```diff
@@ -85,27 +85,66 @@ object YamlDecoder extends YamlDecoderCompanionCrossCompat {
   )
 
   implicit def forInt: YamlDecoder[Int] = YamlDecoder { case s @ ScalarNode(value, _) =>
-    Try(java.lang.Integer.decode(value.replaceAll("_", "")).toInt).toEither.left
+    val normalizedValue =
+      if (value.startsWith("0o")) value.stripPrefix("0o").prepended('0') else value
+
+    Try(java.lang.Integer.decode(normalizedValue.replaceAll("_", "")).toInt).toEither.left
       .map(ConstructError.from(_, "Int", s))
   }
 
   implicit def forLong: YamlDecoder[Long] = YamlDecoder { case s @ ScalarNode(value, _) =>
-    Try(java.lang.Long.decode(value.replaceAll("_", "")).toLong).toEither.left
+    val normalizedValue =
+      if (value.startsWith("0o")) value.stripPrefix("0o").prepended('0') else value
+
+    Try(java.lang.Long.decode(normalizedValue.replaceAll("_", "")).toLong).toEither.left
       .map(ConstructError.from(_, "Long", s))
   }
 
+  private val infinityRegex = """([-+]?)(\.inf|\.Inf|\.INF)""".r
```

Comment:
Do we need a regex? toLowerCase.endsWith etc. wouldn't be enough here?

Reply:
according to spec `.INf` is not a valid`Infinity` :[

also it would be something like:
```scala
if value.nonEmpty then
  if value(0) == '-' then
    if value(1) == '.' then
      if value(2) == 'i' || value(2) == 'I' ...
 ```
which is basically the same thing as what regex would do.

Reply:
Ok sure, that was just a nitpick